appveyor.yml

```
@@ -48,43 +48,36 @@ install:
 - cmd: git submodule update --init --recursive
 - cmd: |-
     cd scripts
-    C:\\Python35\\python set_version_to.py -v %APPVEYOR_BUILD_VERSION% -t %APPVEYOR_BUILD_VERSION%
+    C:\\Python311\\python set_version_to.py -v %APPVEYOR_BUILD_VERSION% -t %APPVEYOR_BUILD_VERSION%
     cd ..
     git --version
 
 
 # to run your custom scripts instead of automatic MSBuild
 build_script:
 - ps: |
-    $isBuildingPR = $false;
-    # if building a temporary merge with master, soft reset to the PR commit so the build contains the PR's hash instead of the merge-commit's hash
+    $buildArgs = '/p:ContinuousIntegrationBuild=true';
+    # if building a temporary merge with master, override the hash with the PR commit's so the build contains the PR's hash instead of the merge-commit's hash
     if ($env:APPVEYOR_PULL_REQUEST_HEAD_COMMIT -and ($env:APPVEYOR_REPO_COMMIT -ne $env:APPVEYOR_PULL_REQUEST_HEAD_COMMIT))
     {
-        git reset --soft "$env:APPVEYOR_PULL_REQUEST_HEAD_COMMIT" --
-        $isBuildingPR = $true;
+        $gitCommit = $(git rev-parse --short $env:APPVEYOR_PULL_REQUEST_HEAD_COMMIT)
+        $buildArgs +=  " /p:GitCommit=$gitCommit /p:GitSha=$($env:APPVEYOR_PULL_REQUEST_HEAD_COMMIT)"
     }
+- ps: |
+    # add to index, to adjust crlf configured in .gitattributes
+    git -c core.autocrlf=false add -A
+    if ($LastExitCode -ne 0) { $host.SetShouldExit($LastExitCode) }
 - ps: |
     # build VC++
     dotnet build .\scripts\native.proj -c Release --verbosity q --nologo /bl:.\artifacts\log\native.binlog
     if ($LastExitCode -ne 0) { $host.SetShouldExit($LastExitCode) }
 - ps: |
     # build .NET
-    dotnet build -c Release --verbosity q --nologo /bl:.\artifacts\log\build.binlog /p:ContinuousIntegrationBuild=true
+    $cmd = "dotnet build -c Release --verbosity q --nologo /bl:.\artifacts\log\build.binlog $buildArgs"
+    iex $cmd
```

```suggestion
    Invoke-Expression $cmd
```
Why is it needed at all?